Support NCSI and GT CX7 NCSI init

Victor-Jhong opened the pull request.

Victor-Jhong committed:
support_ncsi_support_gt_cx7_ncsi_init

Commit:
common: add sensor driver nct7363 (#1734)

Summary:
- add nct7363.c
- add nct7363.h

Description:
- for AALC project sensor

Pull Request resolved: https://github.com/facebook/OpenBIC/pull/1734

Test Plan: - Waiting for test

Reviewed By: wangx6f

Differential Revision: D57047402

fbshipit-source-id: 65f089f3dc20f1dc1bda55f5b4f9462b32515dc2

Eddie043 commented on common/service/ncsi/ncsi.c:
```diff
@@ -0,0 +1,437 @@
+	}
+
+	mctp *mctp_inst = (mctp *)mctp_p;
+	if (!mctp_inst->ncsi_inst_table) {
```
Redundant

Eddie043 commented on common/service/ncsi/ncsi.c:
```diff
@@ -0,0 +1,437 @@
+		LOG_ERR("Inatant id %d not register yet!", inst_num);
+		return false;
+	}
+	WRITE_BIT(mctp_inst->ncsi_inst_table[cur_inst_table], cur_inst_num % 32, 0);
```
cur_inst_num already less than 32

Commit:
common: add hsc enable function in adm1272 driver (#1738)

Summary:
- add bool enable_adm1272_hsc() in nct214.c
- add bool enable_adm1272_hsc() in nct214.h

Description:
- for AALC project enable hsc function

Pull Request resolved: https://github.com/facebook/OpenBIC/pull/1738

Test Plan: - Waiting for test

Reviewed By: wangx6f

Differential Revision: D57090997

fbshipit-source-id: cfe51b6dbda37cbd9e904b3dd5ea4689ceb4eea9

RickyWu-wiwynn committed:
yv4: wf: revise the supported_thresholds byte for PLDM sensors (#1742)

Summary:
# Description:
Correct the supported_thresholds byte.
Revise threshold for some temperature sensors.

# Motivation:
Correct the supported_thresholds byte becuase the PLDM service in BMC will use supported_thresholds to create the threshold interface for PLDM numeric sensors.

Pull Request resolved: https://github.com/facebook/OpenBIC/pull/1742

Test Plan:
Check the PLDM sensors' threshold are correct.

# Test logs
root@bmc:~# mfg-tool sensor-display
    ...
    "WF_1OU_BOARD_INLET_TEMP_C_1_32": {
        "critical": {
            "high": 50.0
        },
        "max": 255.0,
        "min": 0.0,
        "status": "ok",
        "unit": "DegreesC",
        "value": 39.0
    },
    "WF_CXL1_CNTR_TEMP_C_2_32": {
        "critical": {
            "high": 85.0
        },
        "max": 255.0,
        "min": 0.0,
        "status": "ok",
        "unit": "DegreesC",
        "value": 42.0
    },
    "WF_CXL2_CNTR_TEMP_C_3_32": {
        "critical": {
            "high": 85.0
        },
        "max": 255.0,
        "min": 0.0,
        "status": "ok",
        "unit": "DegreesC",
        "value": 48.0
    },
    "WF_ASIC1_DIMM_A_TEMP_C_12_32": {
        "critical": {
            "high": 85.0
        },
        "max": 0.0,
        "min": 0.0,
        "status": "unavailable",
        "unit": "DegreesC",
        "value": null
    },
    "WF_ASIC1_DIMM_B_TEMP_C_13_32": {
        "critical": {
            "high": 85.0
        },
        "max": 0.0,
        "min": 0.0,
        "status": "unavailable",
        "unit": "DegreesC",
        "value": null
    },
    "WF_ASIC1_DIMM_C_TEMP_C_14_32": {
        "critical": {
            "high": 85.0
        },
        "max": 0.0,
        "min": 0.0,
        "status": "unavailable",
        "unit": "DegreesC",
        "value": null
    },
    "WF_ASIC1_DIMM_D_TEMP_C_15_32": {
        "critical": {
            "high": 85.0
        },
        "max": 0.0,
        "min": 0.0,
        "status": "unavailable",
        "unit": "DegreesC",
        "value": null
    },
    "WF_ASIC2_DIMM_A_TEMP_C_16_32": {
        "critical": {
            "high": 85.0
        },
        "max": 0.0,
        "min": 0.0,
        "status": "ok",
        "unit": "DegreesC",
        "value": 51.25
    },
    "WF_ASIC2_DIMM_B_TEMP_C_17_32": {
        "critical": {
            "high": 85.0
        },
        "max": 0.0,
        "min": 0.0,
        "status": "ok",
        "unit": "DegreesC",
        "value": 52.25
    },
    "WF_ASIC2_DIMM_C_TEMP_C_18_32": {
        "critical": {
            "high": 85.0
        },
        "max": 0.0,
        "min": 0.0,
        "status": "ok",
        "unit": "DegreesC",
        "value": 51.75
    },
    "WF_ASIC2_DIMM_D_TEMP_C_19_32": {
        "critical": {
            "high": 85.0
        },
        "max": 0.0,
        "min": 0.0,
        "status": "ok",
        "unit": "DegreesC",
        "value": 52.5
    },
    ...

Reviewed By: wangx6f

Differential Revision: D57097024

fbshipit-source-id: 5bb19de527c56b943389acd02d60f7a35904da2d

matt-lzy committed:
common: patch: add and modify modbus server function (#1739)

Summary:
1. support modbus callback function that function code by user defined.
2. add modbus callback function: multiple write/read holding registers.
3. add param to set modbus uart parity none stop bit to 1
4. modbus serial initialize: enlarge RTU timeout setting (maximum inter-frame delay)

Description:
implement for modbus function in AALC project

Pull Request resolved: https://github.com/facebook/OpenBIC/pull/1739

Test Plan:
1. apply patch and build code successfully.
2. test modbus communication pass: ast1030 demo board(server) & Wedge400 OpenBMC(client)

Reviewed By: wangx6f

Differential Revision: D57091997

fbshipit-source-id: dbf289bd5542757efd8acf22fa493210fadf204a

RickyWu-wiwynn committed:
yv4: sd: fix the issue that BMC couldn't get EID of WF BIC (#1740)

Summary:
# Description:
Fix the issue that BMC couldn't get EID of WF BIC and CXL.

# Motivation:
Revise the function so SD BIC could set EID to WF correctly.

Pull Request resolved: https://github.com/facebook/OpenBIC/pull/1740

Test Plan:
1. Check BMC could get the EID of WF BIC and CXLs.

# Test logs:
1. Check BMC could get the EID of WF BIC and CXLs

root@bmc:~# busctl tree xyz.openbmc_project.MCTP
└─ /xyz
  └─ /xyz/openbmc_project
    └─ /xyz/openbmc_project/mctp
      └─ /xyz/openbmc_project/mctp/1
        ...
        ├─ /xyz/openbmc_project/mctp/1/50
        ├─ /xyz/openbmc_project/mctp/1/52
        ├─ /xyz/openbmc_project/mctp/1/54
        ├─ /xyz/openbmc_project/mctp/1/55
        ...

root@bmc:~# pldmtool base GetTID -m 50
{
    "Response": 50
}
root@bmc:~# pldmtool base GetTID -m 52
{
    "Response": 52
}

Reviewed By: wangx6f

Differential Revision: D57096756

fbshipit-source-id: 38d7e6301bfba3cd060c332b805cb899b0abaf2b

RickyWu-wiwynn committed:
yv4: sd: revise the supported_thresholds byte for PLDM sensors (#1741)

Summary:
# Description:
Correct the supported_thresholds byte.
Revise threshold for some temperature sensors.

# Motivation:
Correct the supported_thresholds byte becuase the PLDM service in BMC will use supported_thresholds to create the threshold interface for PLDM numeric sensors.

Pull Request resolved: https://github.com/facebook/OpenBIC/pull/1741

Test Plan:
Check the PLDM sensors' threshold are correct.

# Test logs
root@bmc:~# mfg-tool sensor-display
    ...
    "MB_INLET_TEMP_C_1_30": {
        "critical": {
            "high": 60.0
        },
        "max": 255.0,
        "min": 0.0,
        "status": "ok",
        "unit": "DegreesC",
        "value": 39.0
    },
    "MB_OUTLET_TEMP_C_2_30": {
        "critical": {
            "high": 80.0
        },
        "max": 255.0,
        "min": 0.0,
        "status": "ok",
        "unit": "DegreesC",
        "value": 40.0
    },
    "MB_FIO_TEMP_C_3_30": {
        "critical": {
            "high": 40.0
        },
        "max": 255.0,
        "min": 0.0,
        "status": "ok",
        "unit": "DegreesC",
        "value": 32.0
    },
    "MB_CPU_TEMP_C_4_30": {
        "critical": {
            "high": 90.0
        },
        "max": 200.0,
        "min": 0.0,
        "status": "unavailable",
        "unit": "DegreesC",
        "value": null
    },
    "MB_DIMM_A_TEMP_C_5_30": {
        "critical": {
            "high": 85.0
        },
        "max": 255.0,
        "min": 0.0,
        "status": "unavailable",
        "unit": "DegreesC",
        "value": null
    },
    "MB_SSD_BOOT_TEMP_C_17_30": {
        "critical": {
            "high": 77.0
        },
        "max": 255.0,
        "min": 0.0,
        "status": "unavailable",
        "unit": "DegreesC",
        "value": null
    },
    "MB_SSD_DATA_TEMP_C_18_30": {
        "critical": {
            "high": 77.0
        },
        "max": 255.0,
        "min": 0.0,
        "status": "unavailable",
        "unit": "DegreesC",
        "value": null
    },
    "MB_VR_CPU0_TEMP_C_19_30": {
        "critical": {
            "high": 100.0
        },
        "max": 255.0,
        "min": 0.0,
        "status": "ok",
        "unit": "DegreesC",
        "value": 45.0
    },
    "MB_VR_SOC_TEMP_C_20_30": {
        "critical": {
            "high": 100.0
        },
        "max": 255.0,
        "min": 0.0,
        "status": "ok",
        "unit": "DegreesC",
        "value": 45.0
    },
    ...

Reviewed By: wangx6f

Differential Revision: D57096956

fbshipit-source-id: e4e4fca602574823e14faebd38053d5eb7d7c6ff

Eddie043 commented on meta-facebook/gt-cc/src/platform/plat_ncsi.c:
```diff
@@ -0,0 +1,80 @@
+	}
+
+	int resp_len = sizeof(struct mellanox_set_self_recovery_setting_resp);
+	uint8_t resp_buf[resp_len];
```
Why not declare "struct mellanox_set_self_recovery_setting_resp resp" directly?

Victor-Jhong committed:
ncsi_fix_redundant_code

Victor-Jhong committed:
change_declare_of_mellanox_set_self_recovery_setting_resp

Victor-Jhong committed:
change_declare_of_mellanox_set_self_recovery_setting_resp_again

Victor-Jhong committed:
Merge branch 'facebook:main' into ncsi_develop
